src/filesystem.rs

```diff
@@ -319,11 +320,13 @@ impl<'a, R: ReadSeek> SquashfsReadFile<'a, R> {
             None => return Ok(()),
         };
         if block.uncompressed {
-            self.raw_data.decompress(&block, &mut self.buf_decompress)?;
-        } else {
             //data is already decompress, so just swap the read and decompress
             //buffers, so the buf_decompress contains the final data.
             std::mem::swap(&mut self.buf_read, &mut self.buf_decompress);
+            self.buf_decompress =
```

Comment:
@rbran  Found the problem! We need to use `block_offset` correctly to read the right start position.

Any way of getting rid of the `.to_vec()` tho ?  :(

Reply:
Is possible to make this operation directly on the Vec using drain, like in https://github.com/wcampbell0x2a/backhand/pull/147